CONTRIBUTING.md

```diff
@@ -16,6 +16,14 @@ The first step is to fork the repository to your namespace and create a branch o
 
 That branch may end up containing one of more commits that are constituting the full scope of the pull request.
 
+### Release Readiness
+
+When considering a contribution, we use the following rule of thumb: **all code in `components/` and `utils/` on the `main` branch must be ready for release at any time.**
+
+Practically, this means that new components or improvements to existing components should not be merged until they meet all requirements of code quality (see the checklist below).
+
+If working on a larger project, consider starting it in the `experimental/` directory. We allow contributions to that directory even if they don't yet meet all of our code quality requirements. Once finished, the code can be moved from `experimental/` into `components/` or `utils/` as a separate pull request.
+
```

Comment:
Perhaps we should also mention that `utils/` crates may follow different release cadences -- "ICU4X vX" does not need to include version bumps to `utils` crates, and they should typically be published whenever we add new functionality instead of waiting for an ICU4X release.

Reply:
Added a new paragraph.